bundles/org.openhab.binding.bondhome/src/main/java/org/openhab/binding/bondhome/internal/handler/BondBridgeHandler.java

```diff
@@ -0,0 +1,299 @@
+        ScheduledFuture<?> lJob = this.listenerJob;
+        if (lJob != null) {
+            lJob.cancel(true);
+            udpListener.shutdown();
+            logger.debug("UDP listener job stopped");
+        }
+    }
+
+    @Override
+    public void childHandlerInitialized(ThingHandler childHandler, Thing childThing) {
+        super.childHandlerInitialized(childHandler, childThing);
+        if (childHandler instanceof BondDeviceHandler) {
+            BondDeviceHandler handler = (BondDeviceHandler) childHandler;
+            synchronized (handlers) {
+                if (handlers.isEmpty()) {
+                    // Start the BPUP update service after the first child device is added
+                    startUDPListenerJob();
+                }
+                if (!handlers.contains(handler)) {
+                    handlers.add(handler);
+                }
+            }
+        }
+    }
+
+    @Override
+    public void childHandlerDisposed(ThingHandler childHandler, Thing childThing) {
+        if (childHandler instanceof BondDeviceHandler) {
+            BondDeviceHandler handler = (BondDeviceHandler) childHandler;
+            synchronized (handlers) {
+                if (handlers.contains(handler)) {
+                    handlers.remove(handler);
+                }
+                if (handlers.isEmpty()) {
+                    // Stop the update service when the last child is removed
+                    stopUDPListenerJob();
+                }
+            }
+        }
+        super.childHandlerDisposed(childHandler, childThing);
+    }
+
+    /**
+     * Forwards a push update to a device
+     *
+     * @param the {@link BPUPUpdate object}
+     */
+    public void forwardUpdateToThing(BPUPUpdate pushUpdate) {
+        BondDeviceState updateState = pushUpdate.deviceState;
+        String topic = pushUpdate.topic;
+        String deviceId = null;
+        if (topic != null) {
+            deviceId = topic.split("/")[1];
+        }
+        // We can't use getThingByUID because we don't know the type of device and thus
+        // don't know the full uid (that is we cannot tell a fan from a fireplace, etc,
+        // from the contents of the update)
+        if (deviceId != null) {
+            synchronized (handlers) {
+                for (BondDeviceHandler handler : handlers) {
+                    String handlerDeviceId = handler.getDeviceId();
+                    if (handlerDeviceId.equalsIgnoreCase(deviceId)) {
+                        handler.updateChannelsFromState(updateState);
+                        break;
+                    }
+                }
+            }
+        } else {
+            logger.warn("Can not read device Id from push update.");
+        }
+    }
+
+    /**
+     * Returns the Id of the bridge associated with the handler
+     */
+    public String getBridgeId() {
+        BondBridgeConfiguration config = getConfigAs(BondBridgeConfiguration.class);
+        return config.bondId;
```

Comment:
Please use the config stored in the field to access the ID, as `getConfigAs()` is an expensive operation. Same for below.

Reply:
changed